Axxes.ToyCollector.Web/Startup.cs

```diff
@@ -1,144 +1,139 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Axxes.ToyCollector.Core.Contracts.Database;
+using Axxes.ToyCollector.Core.Contracts.DependencyResolution.Options;
+using Axxes.ToyCollector.DependencyResolution;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Swashbuckle.AspNetCore.Swagger;
+
+namespace Axxes.ToyCollector.Web
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration, IHostingEnvironment environment)
+        {
+            Configuration = configuration;
+            Environment = environment;
+        }
+
+        public IConfiguration Configuration { get; }
+        public IHostingEnvironment Environment { get; }      
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public IServiceProvider ConfigureServices(IServiceCollection services)
+        {
+            services.Configure<CookiePolicyOptions>(options =>
+            {
+                // This lambda determines whether user consent for non-essential cookies is needed for a given request.
+                options.CheckConsentNeeded = context => true;
+                options.MinimumSameSitePolicy = SameSiteMode.None;
+            });
+
+            var mvcBuilder = 
+                services
+                    .AddMvc()
+                    .SetCompatibilityVersion(CompatibilityVersion.Version_2_1)
+                    .ConfigureApplicationPartManager(LoadRazorViewsFromPlugins);
+
+            // Allows the passing of JSON $type parameters ...
+            mvcBuilder.AddJsonOptions(jsonOptions => jsonOptions.SerializerSettings.TypeNameHandling = TypeNameHandling.Auto);
+
+            LoadAllPlugins(services, mvcBuilder);
+
+            services.Configure<DatabaseConnectionStrings>(Configuration.GetSection("ConnectionStrings"));
+
+            services.AddSwaggerGen(c => { c.SwaggerDoc("v1", new Info{Title = "Toy Collector API", Version = "v1"}); });
+
+            return CreateApplicationServiceProvider(services);
+        }
+
+        private void LoadRazorViewsFromPlugins(ApplicationPartManager apm)
+        {
+            var allRazorViewsDlls = Directory.GetFiles(Path.Combine(Environment.ContentRootPath, "bin"), "Axxes.ToyCollector.Plugins.*.Views.dll", SearchOption.AllDirectories);
+            foreach (string razorViewsDll in allRazorViewsDlls)
+            {
+                apm.ApplicationParts.Add(new CompiledRazorAssemblyPart(Assembly.LoadFrom(razorViewsDll)));
+            }
+        }
+
+        private void LoadAllPlugins(IServiceCollection services, IMvcBuilder mvcBuilder)
```

Comment:
Is this function still necessary? Aren't the controllers also activated with the function above?

Reply:
It is is still needed, the Razor SDK will compile the views into a separate `views.dll` assembly.
![image](https://user-images.githubusercontent.com/2621499/48262210-ffb66f80-e421-11e8-97f0-7dd325cd745b.png)

![image](https://user-images.githubusercontent.com/2621499/48262241-1d83d480-e422-11e8-9a39-03a0d4406236.png)